### PR TITLE
Update no-non-null-assertion.md

### PR DESCRIPTION
**Pull request by TokugawaTakesi**

### CODE REVIEW

**Comment by bbarry** [comment]
Spelling wrong on assertion and general wording is a little odd. Perhaps something like:

* If you don't care about strict null-checking, then you will not need this rule.
* If you must have undefined members for some compilation process, you may not be able to use this rule.

Specific to this case in Vue, undefined appears to be a valid state for those members and shouldn't be asserted.

File: packages/eslint-plugin/docs/rules/no-non-null-assertion.md [hunk 1]
```diff
@@ -28,7 +28,8 @@ const includesBaz: boolean = foo.bar?.includes('baz') ?? false;
 
 ## When Not To Use It
 
-If you don't care about strict null-checking, then you will not need this rule.
+* If you don't care about strict null-checking, then you will not need this rule.
+* Non-null assetion is required in class-based Vue ([https://github.com/kaorun343/vue-property-decorator](vue-property-decorator)) for Vue component properties defining.
```
Comment (bradzacher) [inline packages/eslint-plugin/docs/rules/no-non-null-assertion.md:32]:
```suggestion
* Non-null assertion is required in class-based Vue ([https://github.com/kaorun343/vue-property-decorator](vue-property-decorator)) for Vue component properties defining.
```

**Comment by TokugawaTakesi** [comment]
@bradzacher 

> Why is it required?

Because those are being initialized in life cycle hooks. 
With `strict: true` or `strictPropertyInitialization: true` TypeScript can not detect this.

> None of the examples I see in the readme use a non-null assertion.

Below example is from [documentation](https://github.com/kaorun343/vue-property-decorator#Prop). Currently, non-null assertion is not required for optional properties, but necessary for required properties (`propB`).  Witout non-null assertion, we get  "Property 'propB' has no initialize and is not definitely assigned in the constructor" error.

```
@Component
export default class YourComponent extends Vue {
  @Prop(Number) readonly propA: number | undefined
  @Prop({ default: 'default value' }) readonly propB!: string
  @Prop([String, Boolean]) readonly propC: string | boolean | undefined
}
```

> I see a lot of "definite property definitions", but no non-null assertions.
Please give me the link with example.

**Comment by bradzacher** [comment]
> `readonly propB!: string`

In this case, `!` is not a non-null assertion, it is a "definite property definition".
Here, the `!` means "this property has no initialiser, but trust me; it will be initialised".

Because it's not a non-null assertion, the `no-non-null-assertion` lint rule will not flag this case.

**Comment by TokugawaTakesi** [comment]
You right, I confused the `non-null assertion` and `definite property definition`.
I am sorry about bothering you.

**Pull request closed without merging**